src/target/riscv/riscv-013.c

```diff
@@ -3424,114 +3412,189 @@ static void log_mem_access_result(struct target *target, bool success, int metho
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_DISABLED,                                      \
+			MEM_ACCESS_STATUS_KIND_DISABLED, "disabled")                               \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_SKIPPED_ABSTRACT_ACCESS_CMDERR,                \
+			MEM_ACCESS_STATUS_KIND_SKIPPED, "skipped (abstract access cmderr)")        \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_SKIPPED_PROGBUF_INSUFFICIENT,                  \
+			MEM_ACCESS_STATUS_KIND_SKIPPED, "skipped (insufficient progbuf)")          \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_SKIPPED_UNSUPPORTED_ACCESS_SIZE,               \
+			MEM_ACCESS_STATUS_KIND_SKIPPED, "skipped (unsupported access size)")       \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_SKIPPED_XLEN_TOO_SHORT,                        \
+			MEM_ACCESS_STATUS_KIND_SKIPPED, "skipped (xlen too short)")                \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_SKIPPED_TARGET_NOT_HALTED,                     \
+			MEM_ACCESS_STATUS_KIND_SKIPPED, "skipped (target not halted)")             \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_SKIPPED_TOO_LARGE_ADDRESS,                     \
+			MEM_ACCESS_STATUS_KIND_SKIPPED, "skipped (address too large)")             \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_SKIPPED_UNSUPPORTED_INCREMENT_SIZE,            \
+			MEM_ACCESS_STATUS_KIND_SKIPPED, "skipped (increment size not supported)")  \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_SKIPPED_TARGET_SELECT_FAILED,                  \
+			MEM_ACCESS_STATUS_KIND_SKIPPED, "skipped (dm target select failed)")       \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_SKIPPED_FENCE_EXEC_FAILED,                     \
+			MEM_ACCESS_STATUS_KIND_SKIPPED, "skipped (fence execution failed)")        \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_FAILED,                                        \
+			MEM_ACCESS_STATUS_KIND_FAILED, "failed")                                   \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_FAILED_ABSTRACT_ARG_WRITE_FAILED,              \
+			MEM_ACCESS_STATUS_KIND_FAILED, "failed (abstract argument write failed)")  \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_FAILED_ABSTRACT_ARG_READ_FAILED,               \
+			MEM_ACCESS_STATUS_KIND_FAILED, "failed (abstract argument read failed)")   \
+	MEM_ACCESS_STATUS_HANDLER(MEM_ACCESS_STATUS_FAILED_PRIV_MOD_FAILED,                        \
+			MEM_ACCESS_STATUS_KIND_FAILED, "failed (privilege modification failed)")   \
+
+
+#define MEM_ACCESS_STATUS_HANDLER(name, kind, msg) name,
+typedef enum {
+	LIST_OF_MEM_ACCESS_STATUSES
+} mem_access_status_t;
+#undef MEM_ACCESS_STATUS_HANDLER
+
+bool is_mem_access_failed(mem_access_status_t status)
+{
+	#define MEM_ACCESS_STATUS_HANDLER(name, kind, msg) \
+		case name: return kind == MEM_ACCESS_STATUS_KIND_FAILED;
+	switch (status) {
+		LIST_OF_MEM_ACCESS_STATUSES
+	}
+	#undef MEM_ACCESS_STATUS_HANDLER
+	LOG_ERROR("Unknown memory access status: %d", status);
```

Comment:
If you like:

```suggestion
	assert(false);
```

Or both the assertion and error message can be kept:

```suggestion
	LOG_ERROR("Unknown memory access status: %d", status);
	assert(false);	
```

Reply:
Addressed